packages/flutter_tools/test/commands.shard/permeable/build_bundle_test.dart

```diff
@@ -557,6 +557,238 @@ void main() {
     ProcessManager: () => FakeProcessManager.any(),
   });
 
+  testUsingContext(
+      '--dart-define-from-file successfully forwards values to build env containing equal in value',
+      () async {
+    globals.fs
+        .file(globals.fs.path.join('lib', 'main.dart'))
+        .createSync(recursive: true);
+    globals.fs.file('pubspec.yaml').createSync();
+    globals.fs.file('.packages').createSync();
+    await globals.fs.file('config1.json').writeAsString('''
+        {
+          "myVar": "sfadsfv=432f"
+        }
+      ''');
+    final CommandRunner<void> runner =
+        createTestCommandRunner(BuildBundleCommand(
+      logger: BufferLogger.test(),
+    ));
+
+    await runner.run(<String>[
+      'bundle',
+      '--no-pub',
+      '--dart-define-from-file=config1.json',
+    ]);
+  }, overrides: <Type, Generator>{
+    BuildSystem: () => TestBuildSystem.all(BuildResult(success: true),
+            (Target target, Environment environment) {
+          expect(
+            _decodeDartDefines(environment),
+            containsAllInOrder(const <String>[
+              'myVar=sfadsfv=432f',
+            ]),
+          );
+        }),
+    FileSystem: fsFactory,
+    ProcessManager: () => FakeProcessManager.any(),
+  });
+
+  testUsingContext('test --dart-define-from-file option env file', () async {
```

Comment:
Consider making this test name more consistent with other tests.

```suggestion
  testUsingContext('--dart-define-from-file correctly parses a valid env file', () async {
```